app/controllers/lsg_bodies_controller.rb

```diff
@@ -34,9 +34,17 @@ def update
     redirect_to lsg_body_path(params[:id])
   end
 
-  def destroy
-    authorize LsgBody
-    LsgBody.delete(params[:id])
+  def archive
+    id = params[:lsg_body_id]
+    @lsg_body = LsgBody.find(id)
+    @lsg_body.archive!
+    redirect_to lsg_bodies_path
+  end
+
+  def unarchive
+    id = params[:lsg_body_id]
+    @lsg_body = LsgBody.find(id)
+    @lsg_body.unarchive!
```

Comment:
These endpoints should have an authorized check. 
Use pundit for writing policies for auth check 
https://github.com/varvet/pundit

Reply:
Have rewritten the policy in lsg_bodies_policy. Needs review.